### ISSUE TITLE
ArgumentException when doubleclicking a document

### ISSUE THREAD

**Issue by mjj4791** [dockpanelsuite/dockpanelsuite#600]
I have a document docked in a dockpanel. When I double click on the tab, it becomes a floating window; which is intended

However I get a Argument exception, which causes the application to crash; the form being docked contains a citrix (wfica control)  as well:

How can I prevent / catch this exception from happening?

System.ArgumentException: Parameter is not valid.
   at System.Drawing.Font.ToLogFont(Object logFont, Graphics graphics)
   at System.Drawing.Font.ToLogFont(Object logFont)
   at System.Drawing.Font.ToHfont()
   at System.Windows.Forms.Control.FontHandleWrapper..ctor(Font font)
   at System.Windows.Forms.Control.get_FontHandle()
   at System.Windows.Forms.Control.get_FontHandle()
   at System.Windows.Forms.Control.get_FontHandle()
   at System.Windows.Forms.Control.get_FontHandle()
   at System.Windows.Forms.Control.get_FontHandle()
   at System.Windows.Forms.Control.SetWindowFont()
   at System.Windows.Forms.Control.OnFontChanged(EventArgs e)
   at System.Windows.Forms.AxHost.OnFontChanged(EventArgs e)
   at System.Windows.Forms.Control.OnParentFontChanged(EventArgs e)
   at System.Windows.Forms.Control.OnFontChanged(EventArgs e)
   at System.Windows.Forms.Control.OnParentFontChanged(EventArgs e)
   at System.Windows.Forms.Control.OnFontChanged(EventArgs e)
   at System.Windows.Forms.ContainerControl.OnFontChanged(EventArgs e)
   at System.Windows.Forms.Form.OnFontChanged(EventArgs e)
   at System.Windows.Forms.Control.AssignParent(Control value)
   at System.Windows.Forms.Form.AssignParent(Control value)
   at System.Windows.Forms.Control.ControlCollection.Add(Control value)
   at System.Windows.Forms.Control.set_ParentInternal(Control value)
   at System.Windows.Forms.Form.set_ParentInternal(Control value)
   at System.Windows.Forms.Control.set_Parent(Control value)
   at WeifenLuo.WinFormsUI.Docking.DockContentHandler.SetParent(Control value)
   at WeifenLuo.WinFormsUI.Docking.DockContentHandler.SetPane(DockPane pane)
   at WeifenLuo.WinFormsUI.Docking.DockContentHandler.SetDockState(Boolean isHidden, DockState visibleState, DockPane oldPane)
   at WeifenLuo.WinFormsUI.Docking.DockContentHandler.set_IsFloat(Boolean value)
   at WeifenLuo.WinFormsUI.Docking.DockPaneStripBase.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

**Comment by mjj4791** [comment]
Using DockPanelSuite.3.0.6

**Comment by mjj4791** [comment]
Looks like I was able to find a solution for this!

This issue seems to be caused by the Citrix ICAClient (ActiveX) control not playing nice with Dockpanelsuite; when a form is docked/undocked, the ActiveX control ran into a failure situation and caused the above exception to occur. This exception then caused the application to terminate.

I successfully tested a workaround which works for me, and could work for others as well: https://github.com/mjj4791/AxICAClientWrapper
